lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -750,8 +764,21 @@ void TypeLoweringVisitor::lowerSAWritePath(Operation *op,
   }
 }
 
+static bool
+canLowerConnect(FConnectLike op,
+                PreserveAggregate::PreserveMode aggregatePreservationMode) {
```

Comment:
Does RefType have last connection semantics? If it is guaranteed that there is no mulpitle write into the reftype element, it's safe to not expand connections .

Reply:
I think expanding the connections might be required for two reasons, 
1.  temporary hold for the yet to be lowered `src` and `dest`, because when the connect is lowered, the `src` and `dest` are not yet lowered. It traverses backward.
2.  And, for results of ops like `MemOp` which will not be lowered. So, the debug output port of the `MemOp` will remain a `vector` but the following connects will be lowered, and hence need to be expanded.
Let me know if I'm missing something. It would be great if we need not lower the connects.